protocols/minecraftvanilla.js

```diff
@@ -54,9 +54,10 @@ class MinecraftVanilla extends Core {
                     name: player.name
                 });
             }
-        }
-        for (let i = 0; i < Math.min(json.players.online, 10000); i++) {
-            state.players.push({});
+        } else {
+            for (let i = 0; i < Math.min(json.players.online, 10000); i++) {
```

Comment:
Ah, I believe the original intention here was for i to be initialized to state.players.length. `sample` often doesn't include a full player list, so this loop is meant to fill out the rest of the slots with placeholders.

Reply:
Ah okay, I thought that when sample exists it always includes all players and otherwise we need to use placeholders.

Reply:
@sonicsnes I pushed another commit and also added coments to make this more clear.